Convert to reusable action

iaindillingham opened the pull request:
I've changed multiple files, so it's probably best to step through each commit one-by-one. I've not grouped them especially well, because I didn't want to spend my time resolving merge conflicts 😬. Essentially, though:

- Several commits replace the pre-reusable actions structure/dependencies with the post-reusable actions structure/dependencies, of these notice 8a69108 and 0b210c6
- Several commits add a study as a use case
- Several commits update tooling
- Several commits remove unnecessary functionality and address shortcomings of the tests

Most importantly: This PR is the **minimum** needed to get deciles-charts working as a reusable action. I know you will have excellent suggestions for configuration (e.g. #4). I know you will find the deciles charts suboptimal; the `x` axis tick labels and the legend are truncated, for example. We can address these, and more, in future PRs.

Closes #1

ccunningham101 commented on analysis/deciles_charts.py:
```diff
@@ -0,0 +1,86 @@
+
+
+def get_deciles_chart(measure_table):
+    return charts.deciles_chart(measure_table, period_column="date", column="value")
```
If we have changes to the ebmdatalab decile charts code, do you think we will push updates to the library? Or have our own code here? In relation to [https://github.com/opensafely-core/cohort-extractor/issues/759](url), we might want to be about to output the intermediate deciles tables for output checking.

iaindillingham replied:
I think that we'll probably end up replacing that implementation with a new implementation, in this module. Writing out intermediate deciles tables would be one reason for replacing implementations (in this case, we could revert 31832fa0a2e8caace78ac78d34bf1187f98b1ea6). I decided to fall back to `charts.deciles_chart` because it's the canonical implementation.

ccunningham101 commented on codelists/codelists.txt:
```diff
@@ -0,0 +1,2 @@
+opensafely/ethnicity/2020-04-27
```
Currently we are loading the ethnicity codelist but not using it. For example studies we probably want to avoid circular dependencies of actions using other actions? So might we want to avoid using the cohort joiner in the decile chart action example study? 
Since it is a small example study, it probably would not be too expensive to extract ethnicity for each month.
Or we could assume a reusable action hierarchy in which decile charts could use the cohort joiner, but the cohort joiner would not use decile charts.

iaindillingham commented:
I've removed the ethnicity codelist in 98643d3d1fd824db1cbb7d44acd8d9a18803c7b5 and, because I noticed it, updated the measure ID in cbd9310989584ae6fb7eecd9ec0d7811c6552953. Would you be happy to ✅ the PR, @ccunningham101?
